app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -195,30 +239,54 @@ public void onClick(DialogInterface dialog, int which) {
                 String itemName = editTextItemName.getText().toString().trim();
                 String itemLink = editTextItemLink.getText().toString().trim();
                 String selectedFolderName = spinnerNewPostFolder.getSelectedItem().toString();
-                Integer selectedFolderId = folderIdNameMap.get(selectedFolderName);
-
-                if (itemName == null || itemName == "" || itemLink == null || itemLink == "") {
-                    Toast.makeText(DashboardActivity.this, "Cannot create post, please enter name and link", Toast.LENGTH_LONG).show();
+                String selectedFolderId = folderNameToIdMap.get(selectedFolderName);
+
+                if (itemName == "" || itemLink == null || itemLink == "") {
+                    Toast.makeText(
+                            DashboardActivity.this,
+                            "Cannot create post, please enter name and link",
+                            Toast.LENGTH_LONG
+                    ).show();
                     return;
                 }
 
                 if (!itemLink.matches("^((http|https):\\/\\/)(www.)?[a-zA-Z0-9@:%._\\\\+~#?&/=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%._\\\\+~#?&/=]*)$")) {
-                    Toast.makeText(DashboardActivity.this, "Invalid Link", Toast.LENGTH_LONG).show();
+                    Toast.makeText(
+                            DashboardActivity.this,
+                            "Invalid Link",
+                            Toast.LENGTH_LONG
+                    ).show();
                     return;
                 }
 
                 // Handle post creation logic
-                Call<ResponseBody> call = apiService.createPost("Token " + currentUser.getToken(), new PostRequest(itemName, "", itemLink, selectedFolderId));
+                Call<ResponseBody> call = apiService.createPost(
+                        currentUser.getTokenHeaderString(),
+                        new PostRequest(itemName, "", itemLink, selectedFolderId)
+                );
 
                 call.enqueue(new Callback<ResponseBody>() {
                     @Override
                     public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
-                        if(response.isSuccessful()) {
-                                    // create a local post matching the one created on the API
-                                    currentUser.updateFoldersAndPosts(DashboardActivity.this);
-                                    Toast.makeText(DashboardActivity.this, "Post creation successful.", Toast.LENGTH_LONG).show();
+                        if (response.isSuccessful()) {
+                            // Display success message, then reload Dashboard
```

Comment:
Given the fact that dashboard activity updates folder and post anyways, I think this is great for simplicity